### ISSUE TITLE
Need to clarify or fix `random_state` / `random_seed`

### ISSUE THREAD

**Issue by mvesin** [fedbiomed/fedbiomed#1369]
Need to clarify and/or fix `random_state` / `random_seed`

1. `random_state` / `random_seed` ?

  In  `TorchDataManager` it looks like we configure randomization from `random_state` with

  > `self.rng(self._loader_arguments.get("random_state"))`

  Yet, this `random_state` is only used in very few notebooks where it is part of `model_args`, not of the   `loader_args` which are passed to the `TorchDataManager`

  Q: is it normal or is it some obsolete code that was not converted to using `random_seed` as a `loader_args` ?

2. error in propagation of variables ? 

    When using as documented in doc/tuto/notebooks, variables are not propagated properly

   ```
   model_args = {
     `random_state`: 2222,
   }
   training_args = {
     `random_seed` : 1111,
   }
   ```

   Then in `TorchDataManager.split()`

   ```
   (Pdb) self._loader_arguments
   {'shuffle': True, 'batch_size': 48}
   ```

  Now if we do:

   ```
   training_args = {
         'loader_args': { 'batch_size': 48,
                   'random_seed' : 3333,
                   'random_state': 4444,
                   }, 
   }
   ```

   Then they propagate to data managers `TorchDataManager.split()`

   ```
   (Pdb) self._loader_arguments
   {'shuffle': True, 'batch_size': 48, 'random_seed': 3333, 'random_state': 4444}
   ```

3. code for `TorchDataManager.split()` and `rng()` : simplify and/or comment

    In `TorchDataManager.split()` we call `random_split(...generator=rng())`
    As `rng()` with no arguments always returns None it may be simplified to
    `random_split(...generator=None)`

   Then in `TorchDataManager._init__()` the call to `self._rng = self.rng(self._loader_arguments.get("random_state"))` could be simplified in a 
   
   ```
   rng = self._loader_arguments.get("random_state")
   if rng is not None:
     torch.Generator(device).manual_seed(rng)
    ```

   If these assertions are false (or in complement) some comment could be added to explain 
   the conditions for splitting dataset in `TorchDataManager.split()`

4. same issues for `SkLearnDataManager`

**Comment by mvesin** [comment]
@sharkovsky @srcansiz and all: hopefully adressed point 3 in branch for #1358 #1359 by fixing a *global* seed for pytorch, if provided by the training plan, so it applies to all devices (cpu, gpu) and all operations (shuffle, split).

Checking from the documentation, the previous implementation did not seem to be providing a reproducible split, or even shuffle.

- https://docs.pytorch.org/docs/stable/generated/torch.Generator.html#torch.Generator
- https://docs.pytorch.org/docs/stable/generated/torch.manual_seed.html

Does not attempt to address yet 1. 2. so the `random_state` may not be correctly propagated to `TorchDataManager` (and bugs still exist). This is just "opportunistic" while refactoring `TorchDataManager`.

**Comment by mvesin** [comment]
Additional information regarding 2. : we have 2 different uses of `random_seed`

- in training plan classes `post_init()` we use `training_args['random_seed']` to set the alea used in training plans
- in framework data manager classes `__init__()` we use `loader_args['random_seed']` to set the alea used in data managers / data loaders
